src/common.hpp

```diff
@@ -1,16 +0,0 @@
-#pragma once
```

Comment:
I removed this file because it was unused except in main.cpp, and main.cpp doesn't need most of these dependencies anyway. This simply made the compile time of main.cpp longer for no reason. I added the small set of dependencies main.cpp really had directly into main.cpp.